### PR TITLE
feat(pagination-nav): New navigation based pagination (issue #554)

### PR DESCRIPTION
**Pull request by tmorehouse**
Addresses issue #554 

New pagination component for page navigation

Takes a `base-url`, which then the page number is appended to.

Supports generating `<router-link>` rather than regular links by setting the property `use-router`

### CODE REVIEW

**Comment by mosinve** [comment]
Don't see a catch of this component.
What if i need to  define a custom array of links?
For ex, a typical vue-router config:
```
const routes = [
  { path: '/foo', component: Foo },
  { path: '/bar', component: Bar }
]
```
Is it possible to setup to navigate into these pages, or it just for navigating over enumerable resources, like pages or items?

**Comment by tmorehouse** [comment]
This is for paginating over enumerable pages (say entries in a blog, or pages of an article/story/etc.

I was thinking about the option of providing an array like you suggested, for things that aren't sequential (i.e. 1, 2, 3, 4, 5, etc). But then wan't sure how to display the index number.

Another option might be to modify the baseUrl so that it can take a template (so the page number could be anywhere), or to provide a lookup table that converted the page number (1, 2, 3, etc) into URLs.

For router support, it is currently pretty basic... if `use-router` is set to true, then rather than an HREF ir places the URL into b-link's `to` prop.

Definitely open to suggestions to make this component flexible.(but not too complex to use).

Maybe if an array is provided (maybe a prop called `page-links`), it would use that, and if not st it would use the `number-of-pages` prop and `base-url`

The other option might be to provide a prop that lets people specify a URL/TO generator function (that overrides the built in makeUrl function), and it would be passed the page number and would return either an `href` or `to` prop.

**Comment by mosinve** [comment]
We can add prop `items` just like for table, which can we either an Array, or a Function, but i think this will make component more complex.

**Comment by tmorehouse** [comment]
We could use the idea of a url-maker function (which defaults to the current internal `makeUrl` function), which people could then either return a string (href) or an object (to).  then they can do their own lookup in their own array or whatever to make the individual links.  It would only be called for the displayed page link buttons (and the prev/next buttons).

**Comment by tmorehouse** [comment]
I've added in a `link-gen` function prop, that allows finer grained control over the `href`/`to`

File: lib/components/pagination-nav.vue [hunk 1]
```diff
@@ -90,7 +90,7 @@
             </li>
             <li v-else class="page-item" role="none presentation">
                 <b-link class="page-link"
-                        v-bind="linkProps(currentPage - 1)"
+                        v-bind="linkProps(numberOfPages)"
```
Comment (tmorehouse) [inline lib/components/pagination-nav.vue]: Good catch!   a copy and paste error on my part ;)

Comment (mosinve) [inline lib/components/pagination-nav.vue]: Everything else works fine, linkGen func awsome, but it need detailed docs. That part about using arrays of links and assigning length-1 to total pages, and pagenum-1, especially.

Comment (tmorehouse) [inline lib/components/pagination-nav.vue]: Yeah,  I'll update the docs with a few examples, and explain that page num is a value from 1 to N, while arrays are 0 to N-1.

Comment (tmorehouse) [inline lib/components/pagination-nav.vue]: @mosinve I've now added an example using an array of links to generate the page links.

**Pull request merged**